realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1770,14 +1770,15 @@ public void delete(Class<? extends RealmModel> clazz) {
     }
 
     private void checkHasPrimaryKey(Class<? extends RealmModel> clazz) {
-        String className = configuration.getSchemaMediator().getSimpleClassName(clazz);
-        OsObjectSchemaInfo objectSchemaInfo = sharedRealm.getSchemaInfo().getObjectSchemaInfo(className);
-
-        if (objectSchemaInfo.getPrimaryKeyProperty() == null) {
+        if (!hasPrimaryKey(clazz)) {
             throw new IllegalArgumentException("A RealmObject with no @PrimaryKey cannot be updated: " + clazz.toString());
         }
     }
 
+    boolean hasPrimaryKey(Class<? extends RealmModel> clazz){
```

Comment:
Add minimal documentation and include
```
// Called from proxy classes.
```
like in `createObjectInternal`.

Reply:
I have removed this improvement and will move it to the Mixed datatype as it is not actually required here. We can directly access the metadata instead of using Realm.